app/dashboard/views.py

```diff
@@ -821,6 +794,54 @@ def users_fetch(request):
             fulfilled__bounty__github_url__icontains=organisation
         ).distinct()
 
+    return profile_list
+
+
+
+@require_GET
+def users_fetch(request):
+    """Handle displaying users."""
+    q = request.GET.get('search', '')
+    skills = request.GET.get('skills', '')
+    limit = int(request.GET.get('limit', 10))
+    page = int(request.GET.get('page', 1))
+    order_by = request.GET.get('order_by', '-actions_count')
+    bounties_completed = request.GET.get('bounties_completed', '').strip().split(',')
+    leaderboard_rank = request.GET.get('leaderboard_rank', '').strip().split(',')
+    rating = int(request.GET.get('rating', '0'))
+    organisation = request.GET.get('organisation', '')
+
+    user_id = request.GET.get('user', None)
+    if user_id:
+        current_user = User.objects.get(id=int(user_id))
+    else:
+        current_user = request.user if hasattr(request, 'user') and request.user.is_authenticated else None
+
+    context = {}
+    if not settings.DEBUG:
+        network = 'mainnet'
+    else:
+        network = 'rinkeby'
+    if current_user:
+        profile_list = Profile.objects.prefetch_related(
+                'fulfilled', 'leaderboard_ranks', 'feedbacks_got'
+            ).exclude(hide_profile=True)
+    else:
+        profile_list = Profile.objects.prefetch_related(
+                'fulfilled', 'leaderboard_ranks', 'feedbacks_got'
+            ).exclude(hide_profile=True)
```

Comment:
Oh why did we move this from `def users_fetch_filters` ? to keep that more generic ?

Reply:
To be DRY instead of copy paste the same filters, both users filters and bulk invite user `users_fetch_filters` to filter and have the same result